File: python/program/Motorcycle.py
from Vehicle import Vehicle

class Motorcycle(Vehicle):
    def __init__(self, plat_nomor, merk, tahun_produksi, warna, tipe_motor, kapasitas_tangki):
        super().__init__(plat_nomor, merk, tahun_produksi, warna)
        self._tipe_motor = tipe_motor
        self._kapasitas_tangki = kapasitas_tangki

    def get_tipe_motor(self):
        return self._tipe_motor

    def get_kapasitas_tangki(self):
        return self._kapasitas_tangki

    def set_tipe_motor(self, tipe_motor):
        self._tipe_motor = tipe_motor

    def set_kapasitas_tangki(self, kapasitas_tangki):
        self._kapasitas_tangki = kapasitas_tangki

File: cpp/program/Motorcycle.cpp
#include <iostream>
#include <string>
#include "Vehicle.h"

using namespace std;

class Motorcycle : public Vehicle {
private:
    string tipeMotor;
    int kapasitasTangki;

public:
    Motorcycle(string platNomor, string merk, int tahunProduksi, string warna, string tipeMotor, int kapasitasTangki)
    : Vehicle(platNomor, merk, tahunProduksi, warna), tipeMotor(tipeMotor), kapasitasTangki(kapasitasTangki) {
    }
   
    string getTipeMotor() const {
        return tipeMotor;
    }

    int getKapasitasTangki() const {
        return kapasitasTangki;
    }
   
    void setTipeMotor(const string& newTipeMotor) {
        tipeMotor = newTipeMotor;
    }

    void setKapasitasTangki(int newKapasitasTangki) {
        kapasitasTangki = newKapasitasTangki;
    }
};
